src/main.ts

```diff
@@ -0,0 +1,38 @@
+interface MyType<T> {
+    [key: string]: T;
+}
+
+function Log(target) {
+    console.log('@Name: ', target.name);
+    console.log('@Parameters: ', new target().collection);
+}
+
+@Log
```

Comment:
Decoratorul trebuie sa fie pentru metodele din Map ( getItem, clear, printMap, setItem)

Reply:
Fixed in new commit.
In caz ca nu e in regula pot sa modific.
Merci.